### PR TITLE
Use Gradle's logger to print debug output

### LINKED ISSUE #1855: Don't use println in Gradle plugin
`println` is used in a few places in the Gradle plugin. Instead, use Gradle's `Project.logger()`.

### PR DESCRIPTION
**Pull request by 3flex**
Resolves #1855

### CODE REVIEW

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/ProcessExecutor.kt [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
 import io.gitlab.arturbosch.detekt.BuildFailure
+import org.gradle.api.logging.Logger
 import java.io.BufferedInputStream
 import java.io.BufferedReader
 import java.io.InputStreamReader
 
 object ProcessExecutor {
-    fun startProcess(args: Array<String>, debug: Boolean = false) {
+    fun startProcess(args: Array<String>, logger: Logger) {
```
Comment (schalkms) [inline detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/ProcessExecutor.kt]: Is the Gradle logger a globally available singleton object?
Then we could spare the extra parameter here.

Comment (3flex) [reply detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/ProcessExecutor.kt]: Seems there is - you're right, no point for the parameter in that case.

**Comment by arturbosch** [comment]
I'm sorry I merged this too fast on my mobile without paying attention of the build status. ..
I've reverted the merge. Sorry for the trouble.

**Comment by schalkms** [comment]
@arturbosch 1a6327b causes problems in combination with Jsr223. The reverted commit works fine.

**Pull request merged**